infrastructures/infrastructure-azure/src/main/java/org/ow2/proactive/resourcemanager/nodesource/billing/AzureBillingResourceUsage.java

```diff
@@ -141,24 +141,26 @@ private String queryResourceUsageHistory(String reportedStartTime, String report
     String getLastResourceUsageHistory(AzureBillingCredentials azureBillingCredentials)
             throws IOException, AzureBillingException {
 
-        // Init start date time and end date time
-        // 1. The resource usage start time (the watch time)  will probably not fit
-        // with the "reported date time" (local API server time) since Azure has 19 Data Centers around the world.
-        // To be sure to catch the first resource event, retrieve the resource usage history from yesterday
-        // 2. With hourly granularity Azure only accept start and end date time with '00' set to minutes and seconds (i.e. truncated)
-        // 3. Azure does not accept too recent end date time. Consequently we set end date time to now minus 1 hour.
-        LocalDateTime now = LocalDateTime.now();
-        LocalDateTime nowTruncatedLastHour = now.truncatedTo(ChronoUnit.HOURS);
-        LocalDateTime endDateTime = nowTruncatedLastHour.minusHours(1);
+        // With hourly granularity Azure only accept date times with '00' set to minutes and seconds (i.e. truncated)
```

Comment:
Since those comments are very informative and structured, maybe put them as Method comments... Just saying 😄

Reply:
Good point  but .. this is more a technical comment about the method implementation and how to query azure resources, rather than an explanation of what does this method do or how to use it.

Reply:
I wished I would see more elaborated comments like this one during my reviews !